lib/sort_by_length.rb

```diff
@@ -1,7 +1,40 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(nlog(n)) for merge sort - although bubble sort and insertion sort can have a best-case time complexity of O(n)
+# they have the worst case time complexity of O(n^2) (which is why I chose to do a merge sort).
+# Space complexity: O(n)
 def sort_by_length(my_sentence)
-  raise NotImplementedError, "Method not implemented"
+  array = my_sentence.split(" ")
+  if array.empty?
+    return array
+  else
+    return merge_sort(array)
+  end
 end
+
+# The code below was based on the following merge sort example: https://www.youtube.com/watch?v=Vn-qJijXgf0
+
+def merge_sort(array)
+  if array.length <= 1
+    array
+  else
+    mid_point = (array.length / 2)
+    left = merge_sort(array[0..mid_point - 1])
+    right = merge_sort(array[mid_point..array.length])
+    merge(left, right)
+  end
+
+end
+
+def merge(left, right)
+  if left.empty?
+    right
+  elsif right.empty?
+    left
+  elsif left.first.length <= right.first.length
+    [left.first] + merge(left[1..left.length], right)
```

Comment:
Note since `left[1..left.length]` creates a new array which is a copy of the 1st array (minus 1 element), this **one** line is O(n)